core/resharding.go

```diff
@@ -190,15 +197,15 @@ func (ss *ShardingState) calculateKickoutRate(newNodeList []types.NodeID) float6
 	return math.Max(0.1, math.Min(rate, 1.0))
 }
 
-// getInitShardState returns the initial shard state at genesis.
-func getInitShardState(numberOfShards, numNodesPerShard int) types.ShardState {
+// GetInitShardState returns the initial shard state at genesis.
+func GetInitShardState() types.ShardState {
 	shardState := types.ShardState{}
-	for i := 0; i < numberOfShards; i++ {
+	for i := 0; i < GenesisShardNum; i++ {
```

Comment:
At the very beginning, there are not 3 shards (GenesisShardNum = 3), then the resharding will broke again

Reply:
I added the check to fix the break, but conceptually, there are 3 shards, with 1 shard populated and 2 empty.

Reply:
that's fair enough. please add TODO